src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2153,6 +2155,399 @@ impl Coordinator {
+                Peek(PeekStage),
+            }
+
+            // Array of queries to execute and a function that receives the result rows.
+            let queries: [(
+                AnalyzeQuery,
+                Box<dyn Fn(&mut Vec<String>, Box<dyn RowIterator>) + Send>,
+            ); 4] = [
+                // Hitting persist w/o any filter pushdown. Use `EXPLAIN PLAN INSIGHTS` to find
+                // `sources` with no `op`.
+                (
+                    {
+                        let stage = ExplainStage::PlanInsights;
+                        AnalyzeQuery::Peek(PeekStage::Validate(PeekStageValidate {
+                            plan,
+                            // TODO: Is this the right cluster?
+                            target_cluster,
+                            copy_to_ctx: None,
+                            explain_ctx: ExplainContext::Plan(ExplainPlanContext {
+                                broken,
+                                config: Default::default(),
+                                format: ExplainFormat::Json,
+                                stage,
+                                replan: None,
+                                desc: Some(desc),
+                                optimizer_trace: OptimizerTrace::new(stage.paths()),
+                            }),
+                        }))
+                    },
+                    Box::new(|problems, mut rows| {
+                        while let Some(row) = rows.next() {
+                            let row = row.unpack();
+                            // TODO: Add a paranoia check here that checks this is a str.
+                            let Ok(v) =
+                                serde_json::from_str::<serde_json::Value>(row[0].unwrap_str())
+                            else {
+                                dbg!(&row[0]);
+                                continue;
+                            };
+                            let sources = &v["plans"]["optimized"]["global"]["json"]["sources"];
+                            for source in sources.as_array().into_iter().flatten() {
+                                if source["op"].is_null() {
+                                    problems.push(format!(
+                                        "persist without filter pushdown on {}",
+                                        serde_json::to_string(&source["id"])
+                                            .expect("must serialize")
+                                    ));
+                                }
+                            }
+                        }
+                    }),
+                ),
+                // Source lagging or stalled.
+                (
+                    AnalyzeQuery::Sql(format!(
+                        "
+                        SELECT name, status
+                        FROM mz_internal.mz_source_statuses
+                        WHERE id IN ({dependencies_in}) AND status != 'running'
```

Comment:
Nit: Maybe call some code to grab the string "running", to avoid this going out of sync: https://github.com/MaterializeInc/materialize/blob/795d262c3651096e801fb2b9a22fb972ad7c550e/src/storage-client/src/client.rs#L324